server/src/test/java/org/opensearch/index/ShardIndexingPressureMemoryManagerTests.java

```diff
@@ -0,0 +1,526 @@
+        tracker.getCommonOperationTracker().incrementCurrentCombinedCoordinatingAndPrimaryBytes(1);
+        long requestStartTime = System.currentTimeMillis();
+
+        assertFalse(memoryManager.isCoordinatingShardLimitBreached(tracker, 1 * 1024, requestStartTime));
+        assertFalse(memoryManager.isPrimaryShardLimitBreached(tracker,  1 * 1024, requestStartTime));
+    }
+
+    public void testReplicaShardLimitsNotBreached() {
+        ShardIndexingPressureTracker tracker = memoryManager.getShardIndexingPressureTracker(shardId1);
+        tracker.getReplicaOperationTracker().getStatsTracker().incrementCurrentBytes(1);
+        long requestStartTime = System.currentTimeMillis();
+
+        assertFalse(memoryManager.isReplicaShardLimitBreached(tracker, 1 * 1024, requestStartTime));
+    }
+
+    public void testCoordinatingPrimaryShardLimitsIncreasedAndSoftLimitNotBreached() {
+        ShardIndexingPressureTracker tracker = memoryManager.getShardIndexingPressureTracker(shardId1);
+        tracker.getCommonOperationTracker().incrementCurrentCombinedCoordinatingAndPrimaryBytes(10);
+        long baseLimit = tracker.getPrimaryAndCoordinatingLimits();
+        long requestStartTime = System.currentTimeMillis();
+
+        assertFalse(memoryManager.isCoordinatingShardLimitBreached(tracker, 1 * 1024, requestStartTime));
+        assertFalse(memoryManager.isPrimaryShardLimitBreached(tracker, 1 * 1024, requestStartTime));
+
+        assertTrue(tracker.getPrimaryAndCoordinatingLimits() > baseLimit);
+        assertEquals(tracker.getPrimaryAndCoordinatingLimits(), (long)(baseLimit/0.85));
+    }
+
+    public void testReplicaShardLimitsIncreasedAndSoftLimitNotBreached() {
+        ShardIndexingPressureTracker tracker = memoryManager.getShardIndexingPressureTracker(shardId1);
+        tracker.getReplicaOperationTracker().getStatsTracker().incrementCurrentBytes(15);
+        long baseLimit = tracker.getReplicaLimits();
+        long requestStartTime = System.currentTimeMillis();
+
+        assertFalse(memoryManager.isReplicaShardLimitBreached(tracker, 1 * 1024, requestStartTime));
+        assertTrue(tracker.getReplicaLimits() > baseLimit);
+        assertEquals(tracker.getReplicaLimits(), (long)(baseLimit/0.85));
+    }
+
+    public void testCoordinatingPrimarySoftLimitNotBreachedAndNodeLevelRejections() {
+        ShardIndexingPressureTracker tracker1 = memoryManager.getShardIndexingPressureTracker(shardId1);
+        ShardIndexingPressureTracker tracker2 = memoryManager.getShardIndexingPressureTracker(shardId2);
+        tracker1.getCommonOperationTracker().incrementCurrentCombinedCoordinatingAndPrimaryBytes(4 * 1024);
+        tracker2.compareAndSetPrimaryAndCoordinatingLimits(tracker2.getPrimaryAndCoordinatingLimits(), 6 * 1024);
+        long limit1 = tracker1.getPrimaryAndCoordinatingLimits();
+        long limit2 = tracker2.getPrimaryAndCoordinatingLimits();
+        long requestStartTime = System.currentTimeMillis();
```

Comment:
Use `System.nanoTime` for logical times

Reply:
Sure. I see``System.currentTimeMillis`` frequently used across tests. Given the benefits, other than precision, any specific reason for recommendation?